### ISSUE TITLE
v5.41.0+ Drop-in broken in latest Safari when loading inside an iFrame

### ISSUE THREAD

**Issue by ollym** [Adyen/adyen-web#2149]
When loading a Drop-in within an iFrame. We get the following error in the console:

```
[Error] InvalidAccessError: Trying to start an Apple Pay session from a document with an different security origin than its top-level frame.
```

My expectation is that Apple Pay is just skipped if initialisation fails.

If we add the following to the `/session` endpoint it fixes the issue:

```
blockedPaymentMethods: ['applepay']
```

But means we don't have ApplePay which is not ideal.

**Comment by ollym** [comment]
Just tried v5.42.1 and the same error.

<img width="681" alt="CleanShot 2023-05-09 at 23 28 54@2x" src="https://github.com/Adyen/adyen-web/assets/143771/4670df7d-5a57-4d6f-a13d-c36e3424c764">

Please fix this, we don't want to have apple pay disabled universally for long.

**Comment by ollym** [comment]
PS this appears fixed in the latest webkit:
https://stackoverflow.com/questions/63739348/trying-to-start-an-apple-pay-session-from-a-document-with-an-different-security
https://github.com/WebKit/WebKit/pull/11485

**Comment by ollym** [comment]
I just downloaded the latest Safari Technology Preview and can confirm it works:

<img width="469" alt="CleanShot 2023-05-09 at 23 45 31@2x" src="https://github.com/Adyen/adyen-web/assets/143771/eff83d59-0f1d-4276-9c51-2582215e668b">

You just need to add `allow="payment"` to your parent iframe.

But for now Adyen - you should fix this and gracefully fallback to not offering ApplePay if the initialisation failed.

**Comment by EduardoMateos** [comment]
adyen please fix this. :)

**Comment by ollym** [comment]
@Adyen can someone please acknowledge this issue and its being looked at?

**Comment by ribeiroguilherme** [comment]
Hi @ollym ,

We acknowledged the issue and we are planning to take a look into it in the next weeks. There are other high priority tasks at this moment which the team is busy with, so that is why we couldn't investigate this further. Apologies for the delay

In the meantime: Can you share which version of Safari is this happening?

**Comment by ollym** [comment]
Every version, including iOS

**Comment by ribeiroguilherme** [comment]
And what is your current Safari version that you are testing?

**Comment by ollym** [comment]
`Version 16.4 (18615.1.26.110.1)`

**Comment by ribeiroguilherme** [comment]
@ollym we released 5.45.0 which contains a fix for this issue. Thanks for reaching out to us. cheers!
